design/cassandra/loss-of-persistence.md

```diff
@@ -0,0 +1,69 @@
+
+## Background
+
+### Kubernetes Support For Local Storage
+
+From the beginning of Kubernetes, storage solutions have focused on Network Attached Storage.
+There has been very little support for using local storage and the way to do it until recently was by using hostPath volumes, which had a multitude of problems.
+
+With Kubernetes 1.10, [the local Persistent Volumes feature graduated to beta](https://kubernetes.io/blog/2018/04/13/local-persistent-volumes-beta/), making local storage portable and easy to use on Kubernetes.
+In order to use a local disk as a Persistent Volume, users can leverage [the local volume static provisioner](https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner). 
+To use a local Persistent Volume for a Cassandra Cluster, the user only needs to specify the correct `storageClassName`.
+
+Let's now explore what happens when a local disk fails in Kubernetes, in some different cases:
+1. **Disk exhibits errors and slow performance but hasn't failed yet:** The Local Provisioner won't do something about it. The [local volume monitor](https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/issues/10), a solution being discussed for such problems, should provide a solution once ready.
+2. **Disk fails, mount point disappears:** The Local Provisioner won't do something about it. The [local volume monitor](https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/issues/10), a solution being discussed for such problems, should provide a solution once ready. This condition is easier to detect, so we can look into extending the Local Provisioner if we need it immediately.
+3. **Node with local disk fails:** This is the most common scenario, especially in cloud environments. In a cloud environment, local storage can be lost by VM migrations, Kubernetes or security upgrades, Node failure or fencing.
+
+The 3rd scenario, Node Failure, is the one we will focus on.
+
+
+### Cassandra Support for Member Failures
+
+Cassandra was built with fault tolerance in mind, so hardware failures were expected and included into the design.
+The normal way for a human administrator to deal with a Member failure is:
+
+1. Get the failed Member's IP address.
+2. Start a new Member with the JVM flag `-Dcassandra.replace_address_first_boot=<dead_node_ip>`. When starting, Cassandra will check if the Member has its bootstrap data. If it doesn't, it means that the Member lost its data and should stream it from other Members. 
+
+:warning: This only works if the Replication Factor is high enough. If some data have RF=1 and a Member fails, there is no-one to stream that data from.
+
+## Algorithm
+
+As we established before, our focus will be on the `Node Failure` scenario.
+
+Let's see how a Node Failure would play out.
+Assume a Kubernetes environment with a Cassandra Cluster of 3 Members using local disks, each Member on a different Node.
+Node of member-1 fails, bringing the local disk down with it.
```

Comment:
what happens to the member/pod itself at this point? does it crash too since it doesn't have a functional disk anymore? should the operator be looking for that condition, or should it only take action when it detects a node has been deleted from the cluster?  is the operator already watching nodes in the k8s API, or is that a new requirement?

Reply:
The Node is assumed to have failed, so the Pod is not running.
The deletion of the Node happens by a Cluster Admin.
It is assumed that the Cluster Admin knows the Node has failed or has fenced the Node, before proceeding to delete it.
In other words, the deletion of the Node signifies that it has failed and all resources on it (Pods, PVs) are irrevocably lost.

The Operator doesn't watch Nodes.
When the Operator detects a Member is not Ready, it checks the Member's PVs.
If the Member's PVs have NodeAffinity for a Node that doesn't exist, the Member's PVC is recreated so it can be scheduled again. Otherwise the Pod will be stuck as Pending.